documentation/docs/30-advanced/70-packaging.md

```diff
@@ -126,6 +126,31 @@ This is a legacy field that enabled tooling to recognise Svelte component librar
 }
 ```
 
+### sideEffects
+
+The `sideEffects` option is used by bundlers to determine if a module may contain code that has side effects. A module is considered to have side-effects if it makes changes that are observable from other scripts outside the module when it's imported; such as modifying global variables or the prototype of built-in JavaScript objects. Because a side effect could potentially affect the behavior of other parts of the application, these files/modules will be included in the final bundle regardless of whether their exports are used in the application.
+
+By adding `"sideEffects": false` in `package.json`, you're signaling to the bundler that your package doesn't include any modules with side effects. This information can help the bundler to be more aggressive in eliminating unused exports from the final bundle, a process known as tree-shaking. This results in smaller and more efficient bundles. 
+
+> In the case of a Svelte component library, this prevents CSS from components that are not being used from being included in the final build.
+
+```json
+/// file: package.json
+{
+	"sideEffects": false
+}
+
+Make sure that `"sideEffects"` is correctly set. If a file with side effects is incorrectly marked as having no side effects, it can result in broken functionality. If your package has files with side effects, you can specify them in an array:
+
+```json
+/// file: package.json
+{
+    "sideEffects": ["./src/sideEffectfulFile.js"]
+}
+```
```

Comment:
```suggestion
Different bundlers handle `sideEffects` different. While not necessary for Vite, we recommend that libraries state that all CSS files have side-effects so that your library will be compatible with [webpack](https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free). This information can help the bundler to be more aggressive in eliminating unused exports from the final bundle, a process known as tree-shaking. This results in smaller and more efficient bundles. 

> In the case of a Svelte component library, this prevents CSS from components that are not being used from being included in the final build.

```json
/// file: package.json
{
	"sideEffects": ["**/*.css"]
}

Make sure that `"sideEffects"` is correctly set. If a file with side effects is incorrectly marked as having no side effects, it can result in broken functionality. If your package has files with side effects, you can specify them in an array:

```json
/// file: package.json
{
    "sideEffects": ["**/*.css", "./src/sideEffectfulFile.js"]
}
```
```